packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -145,8 +145,9 @@ export class BlobManager {
         this.logger.sendTelemetryEvent({ eventName: "AttachmentBlobsLoaded", count });
     }
 
-    public snapshot(): ITree {
-        const blobIds = this.runtime.attachState === AttachState.Detached ? this.detachedBlobIds : this.blobIds;
+    public snapshot(attaching?: boolean): ITree {
+        const useDetachedBlobIds = this.runtime.attachState === AttachState.Detached && !attaching;
```

Comment:
should there be pre-condition check:
assert(!attaching || useDetachedBlobIds = this.runtime.attachState === AttachState);

i.e. attaching => detached